docs/source/configuration.mdx

```diff
@@ -124,32 +135,177 @@ server:
+      - headers_remove:
+          matching:
+            regex: ^x-deprecated.*
+      - headers_insert:
+          name: "router-subgraph-name"
+          value: "accounts"
+
+  # Specifies the runtime configuration for talking to the "products" subgraph.
   products:
+    # The URL of the "products" subgraph's GraphQL endpoint
     routing_url: http://localhost:4003/graphql
 
-# graph configuration, see later section
+    # Layers to be applied to the "products" subgraph.
+    layers:
+      - headers_insert:
+          name: "router-subgraph-name"
+          value: "products"
+      - headers_remove:
+          name: "x-legacy-tracking-id"
+
+
+# graph configuration, see documentation below
 # graph:
 
-# spaceport configuration, see later section
+# spaceport configuration, see documentation below
 # spaceport:
 
-# OpenTelemetry configuration, see next section
+# OpenTelemetry configuration, see documentation below
 # opentelemetry:
 ```
 
+## Configuring headers received by subgraphs
+
+Passing headers is supported by leveraging pre-built functionality in the Router known as "layers".  They are therefore defined in the pipeline as members of the named `subgraphs` member (e.g., `reviews`, `products`; often, certain headers only go to certain services) and further defined within the `layers` array.  The `layers` are an array since they're applied in an ordered fashion and later layers may override a previous layer's actions.
+
+There are a few specific layers which offer different header functionality:
+
+- **`headers_propagate`**
+
+  This allows _selectively_ adding headers which were client-originated to the requests made to _specific_ subgraphs.
+
+  They can be configured based on a `matching` pattern using [`regex`]:
+
+  ```yaml
+      - headers_propagate:
+          matching:
+            regex: .*
+  ```
+
+  [`regex`]: https://docs.rs/regex/latest/regex/
+
+  Or by a static string using the `named` option.  These `named` configurations have additional flexibility since they can have either:
+
+  * `default_value`: A value to set in the event that one was not sent by the client.
+  * `rename`: Allow the header's key to be renamed to a new name.
+
+  Additionally, both `default_value` and `rename` can be paired together:
+
+  ```yaml
+      - headers_propagate:
+          named:
+            name: "foo"
+            default_value: "bar"
+            rename: "fii"
+          named:
+            name: "Authorization"
+
+- **`headers_remove`**
+
+  The `headers_remove` layer allows removing a header that was client-originated and not sending it to a specific subgraph.  In the same way as `headers_propagate` can match either exactly or based on a regular expression using [`regex`].
+
+  ```yaml
+      # Do not send this service the "Cookie" header.
+      - headers_remove:
+          name: "Cookie"
+      - headers_remove:
+          matching:
+            # Remove headers using the legacy 'x-' prefix.
+            regex: ^x-.*$
```

Comment:
Suggest putting regexes in quotes to prevent yaml reserved character issues.

Reply:
@BrynCooke I'll probably take this for a follow-up.  A few things that went into my thinking:

1. This wasn't quoted in the example I copied from the PR. 
2. I checked what CircleCI does here [for `filters`](https://circleci.com/docs/2.0/configuration-reference/#jobfilters) and they also didn't quote in their docs. 
3.	  ```yaml
	  # this is valid Yaml and is escaped properly.
	  my_key: this"is a "thing?'
	  ``` 

      I _might_ even argue that adding quotes actually makes things more difficult by forcing it?

4. Spaces are actually Would that be possible with header names?  Header keys [shouldn't have](https://stackoverflow.com/questions/47687379/what-characters-are-allowed-in-http-header-values) (which says, `token` = `CHAR` - `CTL` - `separators`) any characters that need escaping?